keanu-python/keanu/base.py

```diff
@@ -13,9 +13,14 @@ def __repr__(self) -> str:
         return "[{0} => {1}]".format(self._class, type(self))
 
     def __getattr__(self, k: str) -> Callable:
+        self.__check_if_constructed_without_error(k)
         self.__check_if_unwrapped(k)
         raise AttributeError("{} has no attribute {}".format(self.__class__, k))
 
+    def __check_if_constructed_without_error(self, k: str) -> None:
+        if k in ("_val", "_class"):
+            raise ValueError("Object did not get properly constructed - probably due to an earlier unhandled Error.")
+
```

Comment:
this is unrelated to this PR, but it sometimes bites me - typically when a ctor throws an exception, and in the exception there is some kind of error string that ends up calling `__getattr___`

Reply:
How does `_val` and `_class` get called by `__getattr__` in the first place? These are actually valid fields of `JavaObjectWrapper` that we set ourselves.

Reply:
This happens if there's an exception in a constructor of a class that inherits from `JavaObjectWrapper`. In such a case, `_val` and `_class` have not been set.